spec.html

```diff
@@ -28621,6 +28621,191 @@ <h1>
         </emu-clause>
       </emu-clause>
 
+      <emu-clause id="sec-synthetic-module-records">
+        <h1>Synthetic Module Records</h1>
+
+        <p>A <dfn variants="Synthetic Module Records">Synthetic Module Record</dfn> is used to represent information about a module that is defined by specifications. Its exported names are statically defined at creation, while their corresponding values can change over time using SetSyntheticModuleExport. It has no imports or dependencies.</p>
+
+        <emu-note>A Synthetic Module Record could be used for defining a variety of module types: for example, JSON modules or CSS modules.</emu-note>
+
+        <p>In addition to the fields defined in <emu-xref href="#table-module-record-fields"></emu-xref> Synthetic Module Records have the additional fields listed in <emu-xref href="#table-synthetic-module-record-fields"></emu-xref>.</p>
+
+        <emu-table id="table-synthetic-module-record-fields" caption="Additional Fields of Synthetic Module Records">
+          <table>
+            <thead>
+              <tr>
+                <th>Field Name</th>
+                <th>Value Type</th>
+                <th>Meaning</th>
+              </tr>
+            </thead>
+            <tr>
+              <td>[[ExportNames]]</td>
+              <td>a List of Strings</td>
+              <td>The names of the exports of the module. This list does not contain duplicates.</td>
+            </tr>
+            <tr>
+              <td>[[EvaluationSteps]]</td>
+              <td>an Abstract Closure</td>
+              <td>The initialization logic to perform upon evaluation of the module, taking the Synthetic Module Record as its sole argument. It must not modify [[ExportNames]]. It may return an abrupt completion.</td>
+            </tr>
+          </table>
+        </emu-table>
+
+        <emu-clause id="sec-create-default-export-synthetic-module" type="abstract operation">
+          <h1>
+            CreateDefaultExportSyntheticModule (
+              _defaultExport_: an ECMAScript language value,
+            ): a Synthetic Module Record
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>It creates a Synthetic Module Record whose default export is _defaultExport_.</dd>
+          </dl>
+          <emu-alg>
+            1. Let _realm_ be the current Realm Record.
+            1. Let _setDefaultExport_ be a new Abstract Closure with parameters (_module_) that captures _defaultExport_ and performs the following steps when called:
+              1. Perform SetSyntheticModuleExport(_module_, *"default"*, _defaultExport_).
+            1. Return the Synthetic Module Record { [[Realm]]: _realm_, [[Environment]]: ~empty~, [[Namespace]]: ~empty~, [[HostDefined]]: *undefined*, [[ExportNames]]: « *"default"* », [[EvaluationSteps]]: _setDefaultExport_ }.
```

Comment:
In the SMR `Evaluate` operation, `[[EvaluationSteps]]`, when applied to an SMR, is assumed to return a completion record. Yet here, `[[EvaluationSteps]]` is set to `_setDefaultExport_`, which doesn't return anything.

Reply:
`Return NormalCompletion(~unused~).` would be better. (You can't rely on the "Implicit Normal Completion" rule, because the Abstract Closure's return type isn't specified.)